src/tools/rbd_mirror/image_replayer/snapshot/Replayer.cc

```diff
@@ -277,9 +277,16 @@ bool Replayer<I>::get_replay_status(std::string* description,
       matching_remote_snap_it !=
         m_state_builder->remote_image_ctx->snap_info.end()) {
     root_obj["syncing_snapshot_timestamp"] = remote_snap_info->timestamp.sec();
-    root_obj["syncing_percent"] = static_cast<uint64_t>(
-        100 * m_local_mirror_snap_ns.last_copied_object_number /
-        static_cast<float>(std::max<uint64_t>(1U, m_local_object_count)));
+
+    if(!m_local_object_count) {
+      // Set syncing_percent to 0 if m_local_object_count has
+      // not yet been set and the snapshot sync was interrupted.
+      root_obj["syncing_percent"] = 0;
+    } else {
+      root_obj["syncing_percent"] = static_cast<uint64_t>(
+	100 * m_local_mirror_snap_ns.last_copied_object_number /
+	static_cast<float>(std::max<uint64_t>(1U, m_local_object_count)));
```

Comment:
Does this `std::max` logic still make sense in the presence of the new branch?

Reply:
I thought about it but decided to leave it in there in case we change the the if case.

Reply:
Leaving code around "just in case" is bad.  If it's not going to execute, it should be removed.

Also, I'd invert the branches and the condition more obvious (`> 0` is equivalent because `m_local_object_count` is unsigned but reads better):
```
if (m_local_object_count > 0) {
  root_obj["syncing_percent"] = ...
} else {
  root_obj["syncing_percent"] = 0;
}

Reply:
Done.